app/controllers/data_services_controller.rb

```diff
@@ -13,6 +14,11 @@ def show
   private
 
   def data_services
-    @data_services ||= params[:query].blank? ? DataService.all : SearchService.call(query: params[:query])
+    @data_services = DataService.includes(:organisation).all
+    @data_services = SearchService.call(query: params[:query]) if params[:query].present?
```

Comment:
All this logic definitely doesn't belong to the controller. I also suspect you're checking for presence several times. But what all this logic does is basically populating the `@data_services` instance variable, so try to follow the Single Responsibility Principle and have a service object that "just filters" stuff.

- Figure out what data you need in order to filter the dataset. Remember that you can transform the data inside the service object. Passing the whole `params` and "configure" the object in the initialiser is perfectly acceptable
- A Service should be a black box, you just `call` it to have data, hide everything that is not needed to perform the task.
- Reduce the problem to its minimal terms, normally when you're passing around generic `queries` and `filters` you lose the grasp with what you're actually doing: looks like you're filtering by `organisation_id`, `service.name` and `service.description`. A filter should clearly show that.

Few examples of what you're doing: https://thoughtbot.com/blog/a-case-for-query-objects-in-rails
There's also a nice AR feature that might help you: https://www.rubyguides.com/2019/10/scopes-in-ruby-on-rails/